session/session_store.go

```diff
@@ -153,6 +158,20 @@ func (r *RedisStore) BackendIDFromToken(token string) (string, error) {
 	return redis.String(redisConn.Do("HGET", "backend_tokens", token))
 }
 
+// BackendRequiresClientAuth returns a backendID for the token or errors out if none found
+func (r *RedisStore) BackendRequiresClientAuth(backendID string) (bool, error) {
+	redisConn := r.pool.Get()
+	defer redisConn.Close()
+
+	// assume client auth is required unless explicitly disabled
+	noAuth, err := redis.Bool(redisConn.Do("HGET", "backend:"+backendID, "client_auth_disabled"))
+	if err != nil && err == redis.ErrNil {
+		return true, nil
+	}
+
+	return !noAuth, err
```

Comment:
The way error handling is done here I think could be a little dangerous. It certainly works, and is logically correct. But from a readability/future maintainability standpoint, I think it might be better to handle all of the possible non-nil error cases within the first conditional. Eg: Nest all possibilities of returning a non-nil error under the `if err != nil` in line 168.  Just a suggestion since it took me a couple reads to understand that `return !noAuth, err` was indeed the proper behavior. But I'll defer final decision to you on that

Reply:
sure, i can add comments and simplify